Documentation/proposals/20250122-blackbox-crd.md

```diff
@@ -0,0 +1,2722 @@
+## Add BlackboxExporter and BlackboxExporterModule CRDs
+
+* **Owners:**
+  * [Pger-Y](https://github.com/Pger-Y)
+
+* **Related Tickets:**
+  * [Discussion](https://github.com/prometheus-operator/prometheus-operator/discussions/7134#discussioncomment-11569949)
+
+* **Other docs:**
+  * n/a
+
+This document proposes the creation of a Blackbox Operator and associated CRDs, similar to Prometheus-Operator, to manage blackbox-exporter instances and introduce a Module CRD for handling dynamic configuration of blackbox modules in a Kubernetes-native way.
+
+
+
+## Why
+As the above discussion described the Probe CRD is extremely convenient for icmp prober, the Probe CRD can define anything icmp needed ,but when we want to use http Probe , we must define header/body and plenty of something else in blackbox config first,because http is much more complex than icmp.
+
+The situation is very similar to ScrapeConfig/ServiceMonitor that prometheus-operator does,We need a k8s way for generate blackbox config dynamic by reading the Probe Module Config 
```

Comment:
Also I do believe the ScrapeCRD|ServiceMonitor is perhaps not the best comparison, since by design was done to add [additional service discover capabilities to targets outside k8s](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202212-scrape-config.md#why). IMO Alertmanager|AlertmanagerConfig is a better comparison